### DISCUSSION TITLE
Ray tracing LoS

### DISCUSSION THREAD

**Discussion by achaitu** [NVlabs/sionna#237] [Q&A]
Consider a simple situation with a solid obstacle between Tx and Rx (like a big wall of some height and Tx and Rx are present on different sides). Now, I'm trying to compute the received power at the Rx. I'm computing the paths with max_paths=7(this shouldn't matter in this case?) and LoS = True. There will not be any reflectances since there is only one solid wall in the line of sight. For this setting, it is true that just using LoS, without any reflectances, diffractions, or scattering, there should be power that directly comes from the Tx and then attenuated by the wall. 

But I see that no paths exist and if I use paths to compute CIR which I'll then use to compute received power is currently 0. I do not want this behavior. Given any Tx, Rx, Tx will have a direct LoS path (either attenuated by obstacles or not) and we should be able to calculate the received power at the Rx.

**Comment by jhoydis** [comment]
Hi,
You are asking for refraction which Sionna RT currently does not support.

**Reply by achaitu** [reply]
Yeah, I got that, but is there any way to get the signal strength at the Rx in these situations where there is an obstacle between Tx and Rx and Rx does not have any reflected paths from the Tx?

**Reply by jhoydis** [reply]
As I said, Sionna RT does not yet implement refraction. So, if there is no reflected, diffracted, or scattered path that reaches the RX, then the predicted signal strength is 0.

**Reply by achaitu** [reply]
Okay, got it. Say, I do not want the power in this situation to be 0 and since Sionna does not support refraction yet, what is the closest approximation I can do to make the power non-zero at this Rx and give a reasonable value?

1. Is there any way Sionna already stores the distances/path from the Tx to Rx in a non-LoS path? If yes, I could use these already calculated distances (Tx to wall, wall width, wall's other end to Rx) and use some formula to add to the reflections and LoS power value that can be calculated using Sionna. 

Basically, I want a quick and dirty approximation I could do so that the power cannot be 0. For ex, as I said above, I can calculate something along with what Sionna gives and can add both of them ignoring phases and all. This is not perfect, but works for me. What is that I could do?

Thanks.

**Reply by jhoydis** [reply]
In this case, you would need to compute the angles of departure and arrival as well as the distance between the TX and RX, and then compute the path coefficient according to [Equation (26) in the EM Primer](https://nvlabs.github.io/sionna/em_primer.html#equation-h-final). The only missing piece is the transfer matrix $\mathbf{T}_i$ which would be given as

$$
\mathbf{T}_i = \sqrt{g} \frac{e^{-j2\pi\frac{\lambda}{d}}}{d} \mathbf{I}
$$

where $d$ is the distance and $g$ is the path attenuation due to refraction. If you assume that the refraction loss is 10dB, then $g=0.1$. Keep in mind that this is a coarse approximation which does not explicitly account for the thickness and material of the blocking object. However, it should probably be sufficient for your pruposes.

**Reply by Slond** [reply]
Hi there! Is there any plans to support refraction in nearest future?
